libs/parallel_utils.py

```diff
@@ -0,0 +1,41 @@
+import multiprocessing
+import os
+import platform
+from typing import Callable, TypeVar, Iterable, List
+
+from pandarallel import pandarallel
+import pandas as pd
+
+
+VISIBIBLE_PROGRESS_BAR = os.environ.get("PYSEIR_VERBOSITY") == "True"
+pandarallel.initialize(progress_bar=VISIBIBLE_PROGRESS_BAR)
+
+# multiprocessing is unreliable on macOS. See https://bugs.python.org/issue33725#msg343838
+# In theory, using "spawn" start_method would work, but that triggers a bug in pandarallel
+# (https://github.com/nalepae/pandarallel/issues/72).
+USE_MULTIPROCESSING = platform.system() != "Darwin"
+
+T = TypeVar("T")
+R = TypeVar("R")
+SERIES_OR_DF = TypeVar("SERIES_OR_DF", pd.Series, pd.DataFrame)
+
+
+def parallel_map(func: Callable[[T], R], iterable: Iterable[T]) -> List[R]:
+    """Runs func on each item in iterable, in parallel if possible."""
+    if USE_MULTIPROCESSING:
```

Comment:
It might be nice to allow this to be overridden by an environment variable.  I know that I'm generally able to run something like pyseir in parallel on my mac (though not 100% sure how given the issues) and I would be disappointed if i had to change this variable manually to do so.  

May be we should log a warning that it's not being ran in parallel?